File: packages/health_connector_hc_android/android/src/main/kotlin/com/phamtunglam/health_connector_hc_android/mappers/health_measurement_unit_mappers/EnergyMapper.kt
package com.phamtunglam.health_connector_hc_android.mappers.health_measurement_unit_mappers

import androidx.health.connect.client.units.Energy
import com.phamtunglam.health_connector_hc_android.pigeon.EnergyDto

/**
 * Converts an [EnergyDto] to a Health Connect [Energy] object.
 */
internal fun EnergyDto.toHealthConnect(): Energy = Energy.kilocalories(kilocalories)

/**
 * Converts a Health Connect [Energy] object to an [EnergyDto].
 *
 * Uses kilocalories as the transfer unit for consistency.
 */
internal fun Energy.toDto(): EnergyDto = EnergyDto(
    kilocalories = inKilocalories,
)

File: packages/health_connector_hc_android/lib/src/mappers/measurement_unit_mappers/energy_mapper.dart
import 'package:health_connector_core/health_connector_core_internal.dart'
    show Energy;
import 'package:health_connector_hc_android/src/pigeon/health_connector_hc_android_api.g.dart'
    show EnergyDto;
import 'package:meta/meta.dart' show internal;

/// Converts [Energy] to [EnergyDto].
@internal
extension EnergyToDto on Energy {
  EnergyDto toDto() {
    // Uses kilocalories as the transfer unit for consistency.
    return EnergyDto(
      kilocalories: inKilocalories,
    );
  }
}

/// Converts [EnergyDto] to [Energy].
@internal
extension EnergyDtoToDomain on EnergyDto {
  Energy toDomain() {
    return Energy.kilocalories(kilocalories);
  }
}
